test/test_tensorclass.py

```diff
@@ -46,6 +58,7 @@ def test_type():
         y=torch.zeros(3, 4, 5, dtype=torch.bool),
         batch_size=[3, 4],
     )
+    assert isinstance(data, MyDataUndecorated)
```

Comment:
wonderful that it works!

Reply:
I realised that this actually works on `main`, because the undecorated class is a subclass of the dynamically generated class. Updated the test to check that `type(data) is MyDataUndecorated` which is what I was really trying to check for.